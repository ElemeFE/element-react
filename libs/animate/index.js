import Transition from './transition'

export default { Transition }
