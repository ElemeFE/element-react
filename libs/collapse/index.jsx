/* @flow */

import React, { Component } from 'react';

const ANIMATION_DURATION = 300;

type Props = {
  isShow: boolean,
  children?: React.Element<any>
};

export default class CollapseTransition extends Component {
  props: Props;

  selfRef: any;
  leaveTimer: any;
  enterTimer: any;

  componentDidMount(): void {
    this.beforeEnter();
    if (this.props.isShow) {
      this.enter();
    }
  }

  componentWillUnmount(): void {
    this.beforeLeave();
    this.leave();
  }

  componentWillReceiveProps(nextProps: any){
    if (this.props.isShow !== nextProps.isShow) this.triggerChange(nextProps.isShow);
  }

  triggerChange(isShow: boolean): void {
    clearTimeout(this.enterTimer);
    clearTimeout(this.leaveTimer);
    if (isShow) {
      this.beforeEnter();
      this.enter();
    } else {
      this.beforeLeave();
      this.leave();
    }
  }

  beforeEnter(): void {
    const el = this.selfRef;
    //prepare
    el.dataset.oldPaddingTop = el.style.paddingTop;
    el.dataset.oldPaddingBottom = el.style.paddingBottom;
    el.dataset.oldOverflow = el.style.overflow;
    el.style.height = '0';
    el.style.paddingTop = 0;
    el.style.paddingBottom = 0;
  }

  enter(): void {
    const el = this.selfRef;
    //start
    el.style.display = 'block';
    if (el.scrollHeight !== 0) {
      el.style.height = el.scrollHeight + 'px';
      el.style.paddingTop = el.dataset.oldPaddingTop;
      el.style.paddingBottom = el.dataset.oldPaddingBottom;
    } else {
      el.style.height = '';
      el.style.paddingTop = el.dataset.oldPaddingTop;
      el.style.paddingBottom = el.dataset.oldPaddingBottom;
    }

    el.style.overflow = 'hidden';

    this.enterTimer = setTimeout(() => this.afterEnter(), ANIMATION_DURATION);
  }

  afterEnter(): void {
    const el = this.selfRef;
    el.style.display = 'block';
    el.style.height = '';
    el.style.overflow = el.dataset.oldOverflow;
  }

  beforeLeave(): void {
    const el = this.selfRef;
    el.dataset.oldPaddingTop = el.style.paddingTop;
    el.dataset.oldPaddingBottom = el.style.paddingBottom;
    el.dataset.oldOverflow = el.style.overflow;

    el.style.display = 'block';
    if (el.scrollHeight !== 0) {
      el.style.height = el.scrollHeight + 'px';
    }
    el.style.overflow = 'hidden';
  }

  leave(): void {
    const el = this.selfRef;
    if (el.scrollHeight !== 0) {
      el.style.height = 0;
      el.style.paddingTop = 0;
      el.style.paddingBottom = 0;
    }
    this.leaveTimer = setTimeout(() => this.afterLeave(), ANIMATION_DURATION);
  }

  afterLeave(): void {
    const el = this.selfRef;
    if (!el) return ;

    el.style.display = 'none';
    el.style.height = '';
    el.style.overflow = el.dataset.oldOverflow;
    el.style.paddingTop = el.dataset.oldPaddingTop;
    el.style.paddingBottom = el.dataset.oldPaddingBottom;
  }

  render(): React.Element<any> {
    return (
      <div
        className="collapse-transition"
        style={{ overflow: 'hidden' }}
        ref={e => this.selfRef = e}
      >
        {this.props.children}
      </div>
    );
  }
}
