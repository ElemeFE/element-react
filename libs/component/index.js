import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

export default class Component extends React.Component {
  classNames(...args) {
    return classnames(args);
  }

  className(...args) {
    const { className } = this.props;
    return this.classNames.apply(this, args.concat([className]));
  }

  style(args) {
    const { style } = this.props;
    return Object.assign({}, args, style)
  }
}

Component.propTypes = {
  className: PropTypes.string,
  style: PropTypes.object
};
