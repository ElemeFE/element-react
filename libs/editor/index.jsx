import React, { Component } from 'react'
import PropTypes from 'prop-types'
import CodeMirror from 'codemirror'

import 'codemirror/mode/jsx/jsx'
import 'codemirror/keymap/sublime'
import 'codemirror/addon/comment/comment'

import 'codemirror/lib/codemirror.css'
import './style.scss'

export default class Editor extends Component {
  componentDidMount() {
    const { onChange, value } = this.props

    this.cm = CodeMirror(this.editor, {
      mode: 'jsx',
      theme: 'react',
      keyMap: 'sublime',
      viewportMargin: Infinity,
      lineNumbers: false,
      dragDrop: false
    });

    this.cm.setValue(value)

    this.cm.on('changes', cm => {
      if (onChange) {
        clearTimeout(this.timeout);

        this.timeout = setTimeout(() => {
          onChange(cm.getValue());
        }, 300);
      }
    })
  }

  render() {
    return <div className="editor" ref={ref => (this.editor = ref)} />
  }
}

Editor.propTypes = {
  onChange: PropTypes.func,
  value: PropTypes.string
}
