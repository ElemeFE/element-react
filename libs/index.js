/*
 *  Only exposing the libraries for the public components,
 *  Internal components such as Markdown shouldn't be here.
 */

export { default as CollapseTransition } from './collapse';
export { default as Transition } from './transition';
export { default as Component } from './component';
export { default as PropTypes } from './props';
export { default as View } from './view';
