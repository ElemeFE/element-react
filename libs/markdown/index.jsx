import React from 'react';
import ReactDOM from 'react-dom';
import marked from 'marked';
import prism from 'prismjs';

import Canvas from './canvas';

export default class Markdown extends React.Component {
  constructor(props) {
    super(props);

    this.components = new Map;

    this.renderer = new marked.Renderer();
    this.renderer.table = (header, body) => {
      return `<table class="grid"><thead>${header}</thead><tbody>${body}</tbody></table>`;
    };
  }

  componentDidMount() {
    this.renderDOM();
  }

  componentDidUpdate() {
    this.renderDOM();
  }

  renderDOM() {
    for (const [id, component] of this.components) {
      const div = document.getElementById(id);

      if (div instanceof HTMLElement) {
        ReactDOM.unmountComponentAtNode(div);
        ReactDOM.render(component, div);
      }
    }
    prism.highlightAll();
  }

  render() {
    const document = this.document(localStorage.getItem('ELEMENT_LANGUAGE') || 'zh-CN');

    if (typeof document === 'string') {
      this.components.clear();

      const html = marked(document.replace(/:::\s?demo\s?([^]+?):::/g, (match, p1, offset) => {
        const id = offset.toString(36);

        this.components.set(id, React.createElement(Canvas, Object.assign({
          name: this.constructor.name.toLowerCase()
        }, this.props), p1));

        return `<div id=${id}></div>`;
      }), { renderer: this.renderer });

      return (
        <div dangerouslySetInnerHTML={{
          __html: html
        }} />
      )
    } else {
      return <span />
    }
  }
}
