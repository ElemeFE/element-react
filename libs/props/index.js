import PropTypes from 'prop-types';

import rangeType from './range';
import regexType from './regex';

PropTypes.range = rangeType;
PropTypes.regex = regexType;

export default PropTypes;
