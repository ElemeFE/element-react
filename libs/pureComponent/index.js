import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

export default class PureComponent extends React.PureComponent {
  classNames(...args) {
    return classnames(args);
  }

  className(...args) {
    const { className } = this.props;
    return this.classNames.apply(this, args.concat([className]));
  }

  style(args) {
    const { style } = this.props;
    return Object.assign({}, args, style)
  }
}

PureComponent.propTypes = {
  className: PropTypes.string,
  style: PropTypes.object
};
