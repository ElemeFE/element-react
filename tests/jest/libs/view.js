import React from 'react';
import renderer from 'react-test-renderer';

import { View } from '../../../libs'


test('View with one child', () => {
  const component = renderer.create(
    <View>
      <div style={{color: 'white'}} className="white">Children</div>
    </View>
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('View with one child not show', () => {
  const component = renderer.create(
    <View show={false}>
       <div style={{color: 'white'}} className="white">Children</div>
    </View>
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('View with more than one children', () => {
  const component = renderer.create(
    <View style={{color: 'white'}} className="white">
      <div>Children 1</div>
      <div>Children 2</div>
    </View>
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('View with more than one children not show', () => {
  const component = renderer.create(
    <View style={{color: 'white'}} className="white" show={false}>
      <div>Children 1</div>
      <div>Children 2</div>
    </View>
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('View with custom wrap component div', () => {
  const component = renderer.create(
    <View style={{color: 'white'}} className="white" component="div">
      <div>Children 1</div>
      <div>Children 2</div>
    </View>
  );
  expect(component.toJSON()).toMatchSnapshot();
});