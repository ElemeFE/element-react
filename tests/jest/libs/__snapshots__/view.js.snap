// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View with custom wrap component div 1`] = `
<div
  className="white"
  style={
    Object {
      "color": "white",
    }
  }
>
  <div>
    Children 1
  </div>
  <div>
    Children 2
  </div>
</div>
`;

exports[`View with more than one children 1`] = `
<span
  className="white"
  style={
    Object {
      "color": "white",
    }
  }
>
  <div>
    Children 1
  </div>
  <div>
    Children 2
  </div>
</span>
`;

exports[`View with more than one children not show 1`] = `
<span
  className="white"
  style={
    Object {
      "color": "white",
      "display": "none",
    }
  }
>
  <div>
    Children 1
  </div>
  <div>
    Children 2
  </div>
</span>
`;

exports[`View with one child 1`] = `
<div
  className="white"
  style={
    Object {
      "color": "white",
    }
  }
>
  Children
</div>
`;

exports[`View with one child not show 1`] = `
<div
  className="white"
  style={
    Object {
      "color": "white",
      "display": "none",
    }
  }
>
  Children
</div>
`;
